apiAuditLogs/apiAuditLogs.go

```diff
@@ -0,0 +1,277 @@
+		"openDialog":                           true,
+		"patchChannelModerations":              true,
+		"pinPost":                              true,
+		"pushNotificationAck":                  true,
+		"saveReaction":                         true,
+		"searchAllChannels":                    true,
+		"searchArchivedChannelsForTeam":        true,
+		"searchChannelsForTeam":                true,
+		"searchEmojis":                         true,
+		"searchGroupChannels":                  true,
+		"searchPosts":                          true,
+		"searchTeams":                          true,
+		"searchUserAccessTokens":               true,
+		"searchUsers":                          true,
+		"setPostUnread":                        true,
+		"submitDialog":                         true,
+		"teamExists":                           true,
+		"teamMembersMinusGroupMembers":         true,
+		"testElasticsearch":                    true,
+		"testEmail":                            true,
+		"testLdap":                             true,
+		"testS3":                               true,
+		"testSiteURL":                          true,
+		"unpinPost":                            true,
+		"updateUserStatus":                     true,
+		"uploadFileStream":                     true,
+		"viewChannel":                          true,
+	}
+
+	if pass.Pkg.Path() != "github.com/mattermost/mattermost-server/v5/api4" {
+		return nil, nil
+	}
+
+	inspect := pass.ResultOf[inspect.Analyzer].(*inspector.Inspector)
+	nodeFilter := []ast.Node{
+		(*ast.FuncDecl)(nil),
+	}
+	inspect.Preorder(nodeFilter, func(n ast.Node) {
+		funDecl := n.(*ast.FuncDecl)
+
+		if strings.HasSuffix(pass.Fset.File(n.Pos()).Name(), "_test.go") {
+			return
+		}
+		if strings.HasSuffix(pass.Fset.File(n.Pos()).Name(), "apitestlib.go") {
+			return
+		}
+		if obj, ok := pass.TypesInfo.Defs[funDecl.Name].(*types.Func); ok {
+			var fact facts.IsApiHandler
+			if !pass.ImportObjectFact(obj, &fact) {
+				return
+			}
+		}
+		if whiteList[funDecl.Name.Name] {
+			return
+		}
+		initializationFound := false
+		logCallFound := false
+		successCallFound := false
+		ast.Inspect(funDecl, func(node ast.Node) bool {
+			switch n := node.(type) {
+			case *ast.CallExpr:
+				fun, ok := n.Fun.(*ast.SelectorExpr)
+				if !ok {
+					return true
+				}
+
+				instanceType, ok := pass.TypesInfo.Types[fun.X]
+				if !ok {
+					return true
+				}
+
+				if instanceType.Type.String() == "*github.com/mattermost/mattermost-server/v5/audit.Record" && fun.Sel.Name == "Success" {
+					successCallFound = true
+				}
+				if instanceType.Type.String() == "*github.com/mattermost/mattermost-server/v5/web.Context" && (fun.Sel.Name == "LogAuditRec" || fun.Sel.Name == "LogAuditRecWithLevel") {
+					logCallFound = true
+				}
+
+				if instanceType.Type.String() == "*github.com/mattermost/mattermost-server/v5/web.Context" && fun.Sel.Name == "MakeAuditRecord" {
+					initializationFound = true
+					firstArg, ok := n.Args[0].(*ast.BasicLit)
```

Comment:
arg0, arg1 like in the facts file, instead of firstArg, secondArg reads better and shorter.

Reply:
done